src/api-handlers/getProfile.js

```diff
@@ -20,12 +21,31 @@ async function getProfile(fastify, request) {
         legacy = true;
     }
 
-    const result = getProfiles(db, dacConfig, dacId, accounts, legacy);
+    const profiles = await getProfiles(db, dacConfig, dacId, accounts, legacy);
+    const blockedAccounts = await getBlockedAccounts(db, dacId, accounts);
+    const result = { results: [], count: profiles.count };
 
+    profiles.forEach(profile => {
+        const { account } = profile;
+        if (blockedAccounts.includes(account)) {
+            result.results.push(getRedactedCandidateResult(account));
+        } else {
+            result.results.push(profile);
+        }
 
-    return result
+    });
+
+    return result;
 }
 
+const getRedactedCandidateResult = (account) => ({
+  account,
+  error:{
```

Comment:
Front end can check if `results` entry contains `error` object... if yes then they can show proper error title and message. If not then they should grab standard result object with profile data